src/QueryM.purs

```diff
@@ -283,6 +289,30 @@ getDatumsByHashes :: Array DataHash -> QueryM (Map DataHash Datum)
 getDatumsByHashes hashes = unwrap <$> do
   mkDatumCacheRequest DcWsp.getDatumsByHashesCall _.getDatumsByHashes hashes
 
+awaitTxConfirmed :: TxHash -> QueryM Unit
+awaitTxConfirmed = awaitTxConfirmedWithTimeout (Seconds 300.0)
+
+awaitTxConfirmedWithTimeout :: Seconds -> TxHash -> QueryM Unit
+awaitTxConfirmedWithTimeout timeoutSeconds txHash = do
+  nowMs <- getNowMs
+  let timeoutTime = nowMs + unwrap timeoutSeconds * 1000.0
+  go timeoutTime
+  where
+  getNowMs :: QueryM Number
+  getNowMs = unwrap <<< unInstant <$> liftEffect now
+
+  go :: Number -> QueryM Unit
+  go timeoutTime = do
+    isTxFound <- unwrap <$> mkDatumCacheRequest DcWsp.getTxByHash _.getTxByHash
+      txHash
+    nowMs <- getNowMs
+    when (nowMs >= timeoutTime) do
+      liftEffect $ throw $
+        "awaitTxConfirmedWithTimeout: timeout exceeded, Transaction not \
+        \confirmed"
+    liftAff $ delay $ wrap 1000.0
+    if isTxFound then pure unit else go timeoutTime
```

Comment:
The docstring for `Contract.Transaction.awaitTxConfirmed` implies it doesn't have a timeout. I think either it should be updated to say it has a default timeout, or make `awaitTxConfirmedWithTimeout` accept `Maybe Seconds`?

Reply:
`infinity` is a valid `Number`